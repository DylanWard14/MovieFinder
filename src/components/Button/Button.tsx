import React from 'react';
import styled from 'styled-components/macro';
import { darken } from 'polished';

const StyledButton = styled.button`
    background: black;
    border: 1px solid ${props => props.theme.colors.main};
    border-radius: 5px;
    height: 2rem;
    min-width: 4rem;
    color: white;
    margin: 0 1rem;
    font-family: lato;
    font-size: 1rem;

    &:hover {
        cursor: pointer;
        background: ${props => props.theme.colors.main};
        color: black;
    }
    &:active {
        border-color: ${props => darken(0.1, props.theme.colors.main)};
        background: ${props => darken(0.1, props.theme.colors.main)};
    }
`;

export interface ButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {}

const Button: React.FunctionComponent<ButtonProps> = ({ children, ...restProps }) => {
    return <StyledButton {...restProps}>{children}</StyledButton>;
};

export default Button;
