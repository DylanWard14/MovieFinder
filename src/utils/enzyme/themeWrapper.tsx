import React from 'react';
import { render, shallow, mount } from 'enzyme';
import { ThemeProvider } from 'styled-components';

import myTheme from '../../theme';

export const renderWrapper = (component: React.ReactNode) => {
    return render(<ThemeProvider theme={myTheme}>{component}</ThemeProvider>);
};

export const shallowWrapper = (component: React.ReactNode) => {
    return shallow(<ThemeProvider theme={myTheme}>{component}</ThemeProvider>);
};

export const mountWrapper = (component: React.ReactNode) => {
    return mount(<ThemeProvider theme={myTheme}>{component}</ThemeProvider>);
};
