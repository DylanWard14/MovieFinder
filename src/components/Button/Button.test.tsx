import React from 'react';
import { renderWrapper, mountWrapper } from '../../utils/enzyme/themeWrapper';

import Button from './Button';

describe('Button', () => {
    describe('snapshot', () => {
        it('renders correctly', () => {
            const rendered = renderWrapper(<Button onClick={() => {}}>Test</Button>);

            expect(rendered).toMatchSnapshot();
        });
    });

    describe('mounted', () => {
        it('calls onClick when button is clicked', () => {
            const handleOnClick = jest.fn();
            const mounted = mountWrapper(<Button onClick={handleOnClick}>Test</Button>);

            mounted.find('button').simulate('click');
            expect(handleOnClick).toHaveBeenCalled();
        });
    });
});
