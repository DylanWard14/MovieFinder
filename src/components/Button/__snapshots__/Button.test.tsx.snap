// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button snapshot renders correctly 1`] = `
<button
  class="Button__StyledButton-tmhdcf-0 xaYYJ"
>
  Test
</button>
`;
